File: javascript/1221.js
/**
 * @param {string} s
 * @return {number}
 */
var balancedStringSplit = function (s) {
  let m = {
    R: 0,
    L: 0,
  };
  let ans = 0;
  for (val of s) {
    m[val] += 1;

    if (m["R"] == m["L"]) {
      ans += 1;
    }
  }
  return ans;
};

File: go/1221.go
package leetcode

func balancedStringSplit(s string) int {
	m := map[string]int{
		"R": 0,
		"L": 0,
	}
	ans := 0
	for _, val := range s {
		m[string(val)] += 1

		if m["R"] == m["L"] {
			ans += 1
		}
	}
	return ans
}